server/vcs/provider/github/gh_list_iterator.go

```diff
@@ -0,0 +1,39 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	gh "github.com/google/go-github/v45/github"
+	"github.com/palantir/go-githubapp/githubapp"
+	"github.com/pkg/errors"
+	"net/http"
+)
+
+type ListIterator struct {
+	ClientCreator githubapp.ClientCreator
+}
+
+func (i *ListIterator) Iterate(ctx context.Context, installationToken int64, runFunc func(ctx context.Context, client *gh.Client, nextPage int) (interface{}, *gh.Response, error), parseFunc func(interface{}) []string) ([]string, error) {
+	client, err := i.ClientCreator.NewInstallationClient(installationToken)
+	if err != nil {
+		return nil, errors.Wrap(err, "creating installation client")
+	}
+
+	var output []string
```

Comment:
why are you returning []string? I'm sure there are scenarios where we might not want []string back.  Just return a list of interface{} at that point and have the caller decide how to parse.